en/middleware/uxrce_dds.md

```diff
@@ -1,10 +1,10 @@
-# XRCE-DDS (PX4-FastDDS Bridge)
+# uXRCE-DDS (PX4-ROS 2/DDS Bridge)
 
 :::note
```

Comment:
Do you think at some point in this page should we provide context for uXRCE-DDS and the name? 

Specifically, there is 
- a protocol XRCE-DDS. 
- An eProsima implementation of this protocol called "eProsima Micro XRCE-DDS"
- PX4 module uXRCE-DDS that builds on the eProsima code, providing integration with a specified set of uORB topics.
- This is replaceable with other middleware (in theory)? 

You don't need this to do the work, but it might be worth just once making the distinction. Thoughts?

Reply:
That's is a good point, this issue should be addressed so we can clarify which is the "role" of the `uxrce_dds_client` module w.r.t. eProsima Micro XRCE-DDS client library.

It should be something like (a very simple draft), "PX4 adopt the XRCE-DDS agent-client protocol (link) to make its uORB topics available to the DDS network. At the moment PX4 uses eProsima Micro XRCE-DDS (link) to implement that protocol. The agent side (companion computer) is implemented by eProsima Micro XRCE-DDS Agent (link) while the client side is implemented by the `uxrce_dds_client` module (link), which leverages on the eProsima Micro XRCE-DDS client library (link).

In theory yes, we could change middleware, but this would require a new `new-middleware_client` module. This is also one of the reasons the new module name is so similar to the adopted implementation of the protocol.

Reply:
Yes. Suggested modification of your draft - but I'll leave you to PR in.

---

PX4 uses the XRCE-DDS agent-client protocol (link) to make uORB topics available to the DDS network.
The chosen implementation of the protocol is eProsima Micro XRCE-DDS (link).
The PX4 (client-side) is implemented in the `uxrce_dds_client` module (link); this leverages the eProsima Micro XRCE-DDS client library (link) and adds additional code to interwork with uORB topics..
The agent side running on a companion computer is the "vanilla" eProsima Micro XRCE-DDS Agent (link).